app/controllers/posts_controller.rb

```diff
@@ -9,6 +9,34 @@ def show
     @post = @user.posts.find(params[:id])
   end
 
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @user = current_user
+    @post = @user.posts.new(post_params)
+
+    if @post.save
+      redirect_to user_posts_path
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def like
+    @post = Post.find(params[:id]) # Find the post
+    @user = User.find(@post.author_id) # Find the user who authored the post
+
+    @like = @post.likes.new(user: current_user)
+
+    if @like.save
+      redirect_to user_post_path(@user, @post), notice: 'Post liked successfully.'
+    else
+      redirect_to user_post_path(@user, @post), alert: 'Failed to like the post.'
+    end
+  end
```

Comment:
- [ ] **Allow Users to add likes to Posts:**
- Can you host `like` in a different directory(file), which you can name `likes_controller.rb`? I am asking this because it is better to follow MVC best logic: The controller takes data from the Model and prepares it for implementation within the view. Therefore, Using two controllers in one Directory is not the best practice.

Reply:
- Well resolved @Kccendana 👍🏽